### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by wcampbell0x2a**
- Cleanup if statement without !
- Remove `return` in favor or using `else`
- use `.map_or_else()`

### CODE REVIEW

**Comment by jim-signal** [comment]
Hi @wcampbell0x2a, thanks! Could you rebase on our next release, probably 2.9.0? We should be releasing it soon. After that we will look at merging your PR.

**Comment by wcampbell0x2a** [comment]
> Hi @wcampbell0x2a, thanks! Could you rebase on our next release, probably 2.9.0? We should be releasing it soon. After that we will look at merging your PR.

Will do

**Comment by wcampbell0x2a** [comment]
Rebased from `origin/master`, and signed your CLA.

**Review by peter-signal — APPROVED** [review]

File: src/rust/src/webrtc/peer_connection_factory.rs [hunk 1]
```diff
@@ -250,11 +250,10 @@ impl PeerConnectionFactory {
                 hide_ip,
                 ice_servers.rffi(),
                 outgoing_audio_track.rffi(),
-                if let Some(outgoing_video_track) = outgoing_video_track {
-                    outgoing_video_track.rffi()
-                } else {
-                    std::ptr::null()
-                },
+                outgoing_video_track.map_or_else(
+                    || std::ptr::null(),
```
Comment (jim-signal) [inline src/rust/src/webrtc/peer_connection_factory.rs:254]: Can you check this?

```
$ cargo clippy
   Compiling ringrtc v2.9.2 (/ringrtc/src/rust)
warning: redundant closure found
   --> src/webrtc/peer_connection_factory.rs:254:21
    |
254 |                     || std::ptr::null(),
    |                     ^^^^^^^^^^^^^^^^^^^ help: remove closure as shown: `std::ptr::null`
    |
    = note: `#[warn(clippy::redundant_closure)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

warning: 1 warning emitted
```

Comment (wcampbell0x2a) [reply src/rust/src/webrtc/peer_connection_factory.rs:254]: false positive

diff:
```
diff --git a/src/rust/src/webrtc/peer_connection_factory.rs b/src/rust/src/webrtc/peer_connection_factory.rs
index 429cfb1..f2daad5 100644
--- a/src/rust/src/webrtc/peer_connection_factory.rs
+++ b/src/rust/src/webrtc/peer_connection_factory.rs
@@ -250,10 +250,9 @@ impl PeerConnectionFactory {
                 hide_ip,
                 ice_servers.rffi(),
                 outgoing_audio_track.rffi(),
-                outgoing_video_track.map_or_else(
-                    || std::ptr::null(),
-                    |outgoing_video_track| outgoing_video_track.rffi(),
-                ),
+                outgoing_video_track.map_or_else(std::ptr::null(), |outgoing_video_track| {
+                    outgoing_video_track.rffi()
+                }),
                 enable_dtls,
                 enable_rtp_data_channel,
             )
```
error:
```
error[E0277]: expected a `FnOnce<()>` closure, found `*const _`
   --> src/webrtc/peer_connection_factory.rs:253:50
    |
253 |                 outgoing_video_track.map_or_else(std::ptr::null(), |outgoing_video_track| {
    |                                                  ^^^^^^^^^^^^^^^^ expected an `FnOnce<()>` closure, found `*const _`
    |
    = help: the trait `FnOnce<()>` is not implemented for `*const _`
    = note: wrap the `*const _` in a closure with no arguments: `|| { /* code */ }`
```

Comment (jim-signal) [reply src/rust/src/webrtc/peer_connection_factory.rs:254]: You could try without the '()'.

File: src/rust/src/core/call_manager.rs [hunk 2]
```diff
@@ -969,11 +969,12 @@ where
         let mut active_call = check_active_call!(self, "handle_proceed");
         if active_call.call_id() != call_id {
             ringbenchx!(RingBench::CM, RingBench::App, "inactive call_id");
-            return Ok(());
+            Ok(())
+        } else {
+            active_call.set_call_context(app_call_context)?;
+            active_call.inject_proceed(bandwidth_mode)
         }
```
Comment (jim-signal) [inline src/rust/src/core/call_manager.rs:977]: Also, can you remove this empty line?

You could run this to fix the format:
```
cd src/rust
./scripts/format-code
```

See the script for the nightly build that is currently needed.

Also, there are changes in the next release which will run GitHub Actions to see formatting and clippy issues for PRs, so you would know right away if there are any issues like this.

Thanks!

File: src/rust/src/webrtc/peer_connection_factory.rs (hunk 1 above)
Comment (wcampbell0x2a) [reply src/rust/src/webrtc/peer_connection_factory.rs]: True, fixed.

File: src/rust/src/core/call_manager.rs (hunk 2 above)
Comment (wcampbell0x2a) [reply src/rust/src/core/call_manager.rs]: done!

File: src/rust/src/webrtc/peer_connection_factory.rs (hunk 1 above)
Comment (jim-signal) [reply src/rust/src/webrtc/peer_connection_factory.rs]: It looks good, we will merge it to the next release, coming soon. Thanks!

**Comment by jim-signal** [comment]
Merged

**Pull request closed without merging**